### PR TITLE
[AIRFLOW-3765] Fix for create/update xcom values using the admin UI

### PR DESCRIPTION
**Pull request by andyhuynh3**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-3765
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

Refactoring some of the serialization / deserialization logic into a custom SQLAlchemy TypeDecorator so that creating / updating XCom values using the Admin UI works again.

Creating new XCom entry:
image

Updating XCom entry:
image

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason: There already exists XCom tests.

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.
  - All the public functions and the classes in the PR contain docstrings that explain what it does

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

**Comment by ron819** [comment]
Interesting that values in these menus are displayed as not pickled while in other XCOM menu they do:
https://issues.apache.org/jira/browse/AIRFLOW-3094

**Comment by XD-DENG** [comment]
Hi @andyh1203 , based on the screenshots you shared, seems you branched out from a quite old commit (you were using the non-RBAC UI which was already deprecated in commit https://github.com/apache/airflow/commit/c030729dcbe35a2a4869e46b57a761c8376ede8d).

You may want to rebase first to avoid any conflict (anyway there is already a conflict of `airflow/models/__init__.py` in your PR now).

File: airflow/models/__init__.py [hunk 1]
```diff
@@ -188,6 +188,55 @@ def get_fernet():
     return _fernet
 
 
+class XComSerializer(types.TypeDecorator):
```
Comment (seelmann) [inline airflow/models/__init__.py]: I'd suggest to directly move this new class out into it's own file?
See discussion https://lists.apache.org/thread.html/3c15a871b15f99b4dd4bef25085b4de1c167c728f08262df73ef7217@%3Cdev.airflow.apache.org%3E

Comment (andyhuynh3) [reply airflow/models/__init__.py]: Hey @seelmann, thanks for the suggestion. That's a good idea. Wanted to check if it also makes sense to move the XCom model / class to this file as well (i.e. airflow/models/xcom.py would contain the XCom class and XComSerializer class).

**Comment by andyhuynh3** [comment]
Hi @XD-DENG, good catch. Will take that into future consideration! 

On another note, seems like after switching to the RBAC UI, I'm unable to manually create an XCom entry now:
image
Hitting the save button doesn't seem to create a new XCom entry in the metadata db. XCom pushes are still working as expected:
image

**Comment by XD-DENG** [comment]
@andyh1203 there seems to be a potential issue on "manually create a new XCom" feature. I'm checking on it.

On the other hand, I would suggest rebasing and testing your change again. Now actually only RBAC UI is available in master branch.

**Comment by andyhuynh3** [comment]
Having some rebase issues, will close out this PR and reopen once I get things sorted out. Sorry for the trouble.

**Comment by OmerJog** [comment]
@andyh1203 Did you have a chance to revisit this?

**Pull request closed without merging**